rpc/src/lib.rs

```diff
@@ -31,13 +32,29 @@ pub fn internal_err<T: ToString>(message: T) -> Error {
 	}
 }
 
-pub fn handle_call_error((err, data): (sp_runtime::DispatchError, Vec<u8>)) -> Error {
-	let error: &'static str = err.into();
-	// TODO: trim error message
-	let msg = String::from_utf8_lossy(&data);
-	Error {
-		code: ErrorCode::InternalError,
-		message: format!("{}: {}", error.to_lowercase(), msg),
-		data: Some(Value::String(data.to_hex()))
+pub fn error_on_execution_failure(reason: &ExitReason, data: &[u8]) -> Result<(), Error> {
+	match reason {
+		ExitReason::Succeed(_) => Ok(()),
+		ExitReason::Error(e) => {
+			Err(Error {
+				code: ErrorCode::InternalError,
+				message: format!("evm error: {:?}", e),
+				data: Some(Value::String("0x".to_string()))
+			})
+		},
+		ExitReason::Revert(e) => {
+			Err(Error {
+				code: ErrorCode::InternalError,
+				message: format!("evm revert: {:?}", e),
```

Comment:
returned data need to be converted into readable error message

Reply:
Sorry let me fix this in a separate PR!